module4/exercises/06_call_once/01_race.cpp

```diff
@@ -0,0 +1,44 @@
+#include <vector>
+#include <thread>
+#include <iostream>
+#include <sstream> 
+#include <random>
+
+using namespace std;
+
+class Prize {
+    string winnerId{"unknown"};
+    std::random_device dev{};
+    std::mt19937 rng{dev()};  
+    std::uniform_int_distribution<int> dist{10, 50};
+    
+public:
+    void setWinner() {
+        auto id = this_thread::get_id();
+        auto sleepDuration = dist(rng);
+        stringstream msg;
+        msg << "Called " << __FUNCTION__ << "(" << id << "). Chasing time: " << sleepDuration << "ms\n";
+        cout << msg.str(); // single operation on stream is atomic
+        
```

Comment:
```suggestion
        this_thread::sleep_for(chrono::milliseconds(sleepDuration)

```
Brakująca linia - mój błąd.